solr/core/src/java/org/apache/solr/handler/component/DebugComponent.java

```diff
@@ -177,7 +176,7 @@ public void modifyRequest(ResponseBuilder rb, SearchComponent who, ShardRequest
         }
       }
     } else {
-      sreq.params.set(CommonParams.DEBUG_QUERY, "false");
+      sreq.params.set(CommonParams.DEBUG_QUERY, rb.isDebugAll() ? "true" : "false");
```

Comment:
This may look cumbersome, but it is needed in order to be able pass the "debugQuery" parameter to the shard in case that the query is distributed and doesn't return any results.